pkg/publicapi/middleware/version.go

```diff
@@ -80,3 +82,37 @@ func VersionNotifyLogger(logger *zerolog.Logger, serverVersion semver.Version) e
 		},
 	})
 }
+
+// VersionCheckMiddleware returns a middleware that checks if the client version is at least minVersion.
+func VersionCheckMiddleware(serverVersion, minVersion semver.Version) echo.MiddlewareFunc {
+	return func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(c echo.Context) error {
+			clientVersionStr := c.Request().Header.Get(apimodels.HTTPHeaderBacalhauGitVersion)
+			if clientVersionStr == "" ||
+				clientVersionStr == version.DevelopmentGitVersion ||
+				clientVersionStr == version.UnknownGitVersion {
+				// allow the request to pass through
+				return next(c)
+			}
+
+			clientVersion, err := semver.NewVersion(clientVersionStr)
+			if err != nil {
+				return c.JSON(http.StatusBadRequest, map[string]string{
+					"Error": "Invalid client version " + clientVersionStr,
+				})
+			}
+
+			if clientVersion.LessThan(&minVersion) {
+				// Client version is less than the minimum required version
+				return c.JSON(http.StatusForbidden, map[string]string{
+					"Error":          "Client version is outdated. Update your client",
+					"ServerVersion":  serverVersion.String(),
+					"MinimumVersion": minVersion.String(),
```

Comment:
Currently this will return the minimum version is https://github.com/bacalhau-project/bacalhau/pull/4117/files#diff-19907c38cadb733516c5d0d2d452e8ef1dde593253b301251fec43f1a8a8d315R30 (1.3.3) which isn't actually a version. Should we bump the min version to 1.4.0?